### PR TITLE
All the pull requests

### PR DESCRIPTION
**Pull request by bobthecow**
This includes @jaypea's #69, with a couple of changes:

 * Rename the `appendExisting` option to `merge`.
 * Read the existing manifest async instead of sync when merging. 

This also includes @goldibex's #75, but merged into changes from #69 (plus a few minor code style changes).

Thoughts? /cc @sindresorhus @jaypea @goldibex

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -75,22 +75,19 @@ var rev = require('gulp-rev');
...
+	return gulp.src(['assets/css/*.css', 'assets/js/*.js'], {base: 'assets'})
 		.pipe(gulp.dest('build/assets'))
 		.pipe(rev())
 		.pipe(gulp.dest('build/assets'))
-
-		// Add rev-manifest.json as a new src to prevent rev'ing rev-manifest.json
-		.pipe(gulp.src('build/assets/rev-manifest.json', {base: 'assets'}))
-		.pipe(rev.manifest())             // applies only changes to the manifest
+		.pipe(rev.manifest({
+			base: process.cwd() + '/build/assets',
```
Comment (sindresorhus) [inline readme.md]: Why not just `base: 'build/assets',`?

**Comment by sindresorhus** [comment]
> Should we make the manifest backwards compatible by snagging the manifest file if it comes by in the stream?

No

-

Sorry about the delay all. I'd like to merge this a quick as possible. Anything I can do to make it happen.

File: readme.md (hunk 1 above)
Comment (bobthecow) [inline readme.md]: should work.

**Comment by bobthecow** [comment]
> No

kk. i'll make that `base` change. do we want to hold off on a merge until people confirm? it doesn't look like anyone's paying attention to the issue :)

**Comment by sindresorhus** [comment]
Nah, too bad for them if it doesn't solve their use-case. They've had lots of opportunities to give feedback. We can iterate later on if needed.

**Comment by bobthecow** [comment]
base fixed in 7962cf6

**Comment by sindresorhus** [comment]
Alright. Late here, so gonna look it over one last time tomorrow morning and merge then.

**Comment by ijy** [comment]
I'll be able to give this a test tomorrow if it helps. Sorry for the delay, it's been pretty busy of late. This looks like it should solve the problem once and for all. The time and consideration put into it is certainly not unnoticed or unappreciated. Thanks guys. Will kick the tyres tomorrow.

**Comment by bobthecow** [comment]
@sindresorhus one more, from earlier in the thread:

> The readme could use a bit of hierarchy help. Specifically, Integration, Sourcemaps and Works with seem to be at the wrong heading level.

**Comment by bobthecow** [comment]
@ijy Thanks! I hear you, I've been super busy myself :)

**Comment by sindresorhus** [comment]
@ijy Were you able to try it out?

**Comment by montogeek** [comment]
Merge please :)

**Comment by sindresorhus** [comment]
Landed! Thanks all :)

od3vvoq

**Comment by bobthecow** [comment]
Thanks @sindresorhus!

**Comment by austinpray** [comment]
Sweeeeet.

**Comment by SimenB** [comment]
I wasn't able to make this work using ``options.base`` (everytime I used it, the json-file was placed in root). The I looked at the tests, and it used ``options.path``. If I used ``path`` as well I got it to work.

Am I doing something wrong (probably), or is the documentation wrong?

**Comment by bobthecow** [comment]
Yes, that's still the case. See the "This is the right one, but nobody actually knows how path, base and cwd play together in vinyl, so most people prolly won't do this on the first try" :)

**Comment by bobthecow** [comment]
@sindresorhus I think that might be the happiest kitty ever made.

**Pull request closed without merging**